compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -169,39 +174,30 @@ object Inliner {
     /** Removes all Inlined trees, replacing them with blocks.
      *  Repositions all trees directly inside an inlined expansion of a non empty call to the position of the call.
      *  Any tree directly inside an empty call (inlined in the inlined code) retains their position.
+     *
+     *  Until we implement JSR-45, we cannot represent in output positions in other source files.
+     *  So, reposition inlined code from other files with the call position.
      */
     class Reposition extends TreeMap(cpyWithNewSource) {
-      def finalize(tree: Tree, copied: untpd.Tree) =
-        copied.withSpan(tree.span).withAttachmentsFrom(tree).withTypeUnchecked(tree.tpe)
-
-      def reposition(tree: Tree)(implicit ctx: Context): Tree = enclosingInlineds match {
-        case call :: _ if call.symbol.source != curSource =>
-          tree match {
-            case _: EmptyTree[?] | _: EmptyValDef[?] => tree
-            case _ =>
-              // Until we implement JSR-45, we cannot represent in output positions in other source files.
-              // So, reposition inlined code from other files with the call position:
-              tree.withSpan(span)
-          }
-        case _ => tree
-      }
 
       override def transform(tree: Tree)(implicit ctx: Context): Tree = {
-        val transformed = reposition(tree match {
-          case tree: Inlined =>
-            tpd.seq(transformSub(tree.bindings), transform(tree.expansion)(inlineContext(tree.call)))(ctx.withSource(curSource)) : Tree
-          case tree: Ident => finalize(tree, untpd.Ident(tree.name)(curSource))
-          case tree: Literal => finalize(tree, untpd.Literal(tree.const)(curSource))
-          case tree: This => finalize(tree, untpd.This(tree.qual)(curSource))
-          case tree: JavaSeqLiteral => finalize(tree, untpd.JavaSeqLiteral(transform(tree.elems), transform(tree.elemtpt))(curSource))
-          case tree: SeqLiteral => finalize(tree, untpd.SeqLiteral(transform(tree.elems), transform(tree.elemtpt))(curSource))
-          case tree: TypeTree => tpd.TypeTree(tree.tpe)(ctx.withSource(curSource)).withSpan(tree.span)
-          case tree: Bind => finalize(tree, untpd.Bind(tree.name, transform(tree.body))(curSource))
+        def finalize(copied: untpd.Tree) =
+          val span = if tree.source == curSource then tree.span else callSpan
+          copied.withSpan(span).withAttachmentsFrom(tree).withTypeUnchecked(tree.tpe)
```

Comment:
Do we need to change the `tree.source`?

Reply:
That is done by the [`cpyWithNewSource`](https://github.com/lampepfl/dotty/pull/8648/files#diff-021cdae2fbad688e5717101b37e981d1R181). This logic did not change.